### PR TITLE
#1902703: add `select subs` and `try azure for free` for account related errors

### PR DESCRIPTION
**Pull request by wangmingliang-ms**
[AB#1902703](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1902703): add `select subs` and `try azure for free` for account related errors

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/IAccountActions.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.auth;
+
+import com.microsoft.azure.toolkit.lib.common.action.Action;
+
+public interface IAccountActions {
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/IAccountActions.java:10]: why interface here, do we need to implement it later?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java [hunk 2]
```diff
@@ -124,8 +124,7 @@ public void selectSubscription(List<String> selectedSubscriptionIds) {
                 manager.runOnPooledThread(Preloader::load);
             }
         } else {
-            throw new AzureToolkitAuthenticationException("no subscriptions are selected, " +
-                    "make sure you have provided valid subscription list");
+            throw new AzureToolkitRuntimeException("the selected subscriptions are invalid", IAccountActions.SELECT_SUBS, IAccountActions.AUTHENTICATE);
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java]: do we need an new action `RE_AUTHENTICATE`?

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/IAccountActions.java (hunk 1 above)
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/IAccountActions.java:10]: yes, and it's implemented by `IntellijAccountActionsContributor`
We just make some action ids here and expect ide-s to provide these actions. refer to `IdeActions`

File: azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java (hunk 2 above)
Comment (wangmingliang-ms) [reply azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/Account.java]: not sure is re_authenticate is required

**Review by Flanker32 — DISMISSED** [review]
LGTM

File: azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/action/Action.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
 public class Action<D> {
     public static final String SOURCE = "ACTION_SOURCE";
     public static final Id<Runnable> REQUIRE_AUTH = Id.of("action.common.requireAuth");
-    public static final Id<Void> AUTHENTICATE = Id.of("action.common.authenticate");
+    public static final Id<Void> AUTHENTICATE = Id.of("action.account.authenticate");
```
Comment (Flanker32) [inline azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/action/Action.java:37]: may define the AUTHENTICATE in `IAccountActions` and refer it in `Action`

**Review by andxu — APPROVED** [review]
LGTM

**Review by Flanker32 — APPROVED** [review]
LGTM

**Pull request merged**